libs/oracledb/langchain_oracledb/vectorstores/oraclevs.py

```diff
@@ -78,30 +80,38 @@ def _convert_oper_to_sql(oper: str) -> str:
     return oper_map.get(oper, "==")
 
 
-def _generate_condition(condition: FilterCondition) -> str:
-    key = condition["key"]
+def _generate_condition(condition: FilterCondition, bind_variables: List[str]) -> str:
+    key = _quote_indentifier(condition["key"])
     oper = _convert_oper_to_sql(condition["oper"])
     value = condition["value"]
-    if isinstance(value, str):
-        value = f'"{value}"'
-    return f"JSON_EXISTS(metadata, '$.{key}?(@ {oper} {value})')"
+
+    value_bind = f":value{len(bind_variables)}"
+    bind_variables.append(value)
+
+    return (
+        f"JSON_EXISTS(metadata, "
+        f"'$.{key}[*]?(@ {oper} $val)' "
+        f'PASSING {value_bind} AS "val")'
+    )
 
 
-def _generate_where_clause(db_filter: Union[FilterCondition, FilterGroup]) -> str:
+def _generate_where_clause(
+    db_filter: FilterCondition | FilterGroup, bind_variables: List[str]
```

Comment:
we might still need to support simple filter like blow for and (SODA filter allows this apart from $and keyword. This can be considered for future i think... 

`{ category: "books", price: 10 }`